packages/flutter/lib/src/material/button_style.dart

```diff
@@ -435,3 +435,33 @@ class _LerpShapes implements MaterialStateProperty<OutlinedBorder?> {
     return ShapeBorder.lerp(resolvedA, resolvedB, t) as OutlinedBorder?;
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+///
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
```

Comment:
It could be used by other buttons too, maybe say something like `This is used by [TextButton], [ElevatedButton], and [OutlinedButton].` so that it's not implied to be exclusively those?